.github/workflows/tests.yml

```diff
@@ -177,3 +172,99 @@ jobs:
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-${{ matrix.python-version}}-pip-test-${{ hashFiles('setup.py','requirements-dev.txt','constraints.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version}}-pip-test-
+            ${{ runner.os }}-${{ matrix.python-version}}-pip-
+            ${{ runner.os }}-${{ matrix.python-version}}-
+      - name: Install Deps
+        run: |
+          set -e
+          pip install -U -c constraints.txt -r requirements-dev.txt
+      - name: Install Aer
+        run: python -m pip install -U .
+      - name: Run Tests
+        env:
+          QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+        run: |
+          set -e
+          pip check
+          stestr run --slowest
+        shell: bash
+        if: runner.os != 'macOS' || matrix.python-version != '3.8'
+      - name: Run Tests macOS 3.8
+        env:
+          QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+          QISKIT_IN_PARALLEL: TRUE
+        run: |
+          set -e
+          pip check
+          stestr run --slowest
+        shell: bash
+        if: runner.os == 'macOS' && matrix.python-version == '3.8'
+  tests_windows:
+    runs-on: ${{ matrix.os }}
+    needs: ["lint", "sdist"]
+    timeout-minutes: 60
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10"]
+        os: ["windows-2019"]
+    env:
+      AER_THRUST_BACKEND: OMP
+      QISKIT_TEST_CAPTURE_STREAMS: 1
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~\AppData\Local\pip\Cache
+          key: ${{ runner.os }}-${{ matrix.python-version}}-pip-test-${{ hashFiles('setup.py','requirements-dev.txt','constraints.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version}}-pip-test-
+            ${{ runner.os }}-${{ matrix.python-version}}-pip-
+            ${{ runner.os }}-${{ matrix.python-version}}-
+      - name: Add msbuild to PATH
+        uses: microsoft/setup-msbuild@v1.0.2
+      - name: Install Aer Windows
+        run: |
+          set -e
+          python setup.py bdist_wheel -- -G 'Visual Studio 16 2019'
+          pip install --find-links=dist qiskit-aer
+        shell: bash
+      - name: Run Tests
+        env:
+          QISKIT_SUPPRESS_PACKAGING_WARNINGS: Y
+          LANG: 'C.UTF-8'
+          PYTHONIOENCODING: 'utf-8:backslashreplace'
+        run: |
+          set -e
+          chcp.com 65001
+          pip check
```

Comment:
What's this about for Windows?

Reply:
The code page piece is about circuit drawing. The default code page in windows is not utf8 and the text circuit drawer will emit characters the windows character set doesn't understand. This would cause an encoding error when a test errored or tried to print a circuit diagram. Changing the code page to utf8 prior to running the tests solves this.

The pip check line is mostly legacy from the run up to pip having a depsolver. We added it per the general recommendation of the pypa to test how the depsolver would work in our environment prior to the depsolver going live (in pip 21.0 iirc). We can probably remove it but it also doesn't hurt anything it's a quick check

Reply:
it was the code page thing I was really wondering about, yeah.  Good to know, thanks!